static/app/utils/useUpdateOrganization.tsx

```diff
@@ -0,0 +1,74 @@
+import OrganizationStore from 'sentry/stores/organizationStore';
+import type {Organization} from 'sentry/types/organization';
+import {
+  fetchMutation,
+  setApiQueryData,
+  useMutation,
+  useQueryClient,
+  type ApiQueryKey,
+} from 'sentry/utils/queryClient';
+
+interface Variables extends Partial<Organization> {}
+
+type Context =
+  | {
+      previousOrganization: Organization;
+      error?: never;
+    }
+  | {
+      error: Error;
+      previousOrganization?: never;
+    };
+
+function makeDetailedOrganizationQueryKey(organization: Organization): ApiQueryKey {
+  return [`/organizations/${organization.slug}/`];
+}
+
+export function useUpdateOrganization(organization: Organization) {
+  const queryClient = useQueryClient();
+
+  const queryKey = makeDetailedOrganizationQueryKey(organization);
+
+  return useMutation<Organization, Error, Variables, Context>({
+    onMutate: (data: Variables) => {
+      const previousOrganization =
+        queryClient.getQueryData<Organization>(queryKey) ||
+        OrganizationStore.get().organization ||
+        organization;
+
+      if (!previousOrganization) {
+        return {error: new Error('Previous organization not found')};
+      }
+
+      const updatedOrganization = {
+        ...previousOrganization,
+        ...data,
+      };
+
+      // Update caches optimistically
+      OrganizationStore.onUpdate(updatedOrganization);
+      setApiQueryData<Organization>(queryClient, queryKey, updatedOrganization);
```

Comment:
might want a quick test covering OrganizationStore updates unless you're about to use it and cover it there